doc/changelog.md

```diff
@@ -323,10 +355,17 @@
   of `target="mlir"`.
   [(#522)](https://github.com/PennyLaneAI/catalyst/pull/531)
 
-* Only set `JAX_DYNAMIC_SHAPES` configuration option during `trace_to_mlir()`.
+* Fixes a bug in the configuration of dynamic shaped arrays that would cause certain program to
+  error with `TypeError: cannot unpack non-iterable ShapedArray object`.
   [(#526)](https://github.com/PennyLaneAI/catalyst/pull/526)
 
-* Handle run time exception in async qnodes.
+  This is fixed by replacing the code which updates the `JAX_DYNAMIC_SHAPES` option with a
+  `transient_jax_config()`` context manager which temporarily sets the value of
+  `JAX_DYNAMIC_SHAPES` to True and then restores the original configuration value following the
+  yield. The context manager is used by `trace_to_mlir()``.
+
+* Exceptions encountered in the runtime when using the `@qjit` option `async_qnodes=Tue`
+  will now be properly propagated to the frontend.
   [(#447)](https://github.com/PennyLaneAI/catalyst/pull/447)
   [(#510)](https://github.com/PennyLaneAI/catalyst/pull/510)
 
```

Comment:
@dime10 I can't comment on it below, but I'm a bit confused about the `qinst` primitve entry; it says this change is not user facing, so I'm confused what the underlying bug was?

Reply:
The bug was if present if you use some non-user functions, it's not a user-facing change because that functionality was not affected by the bug.